lib/elixir/test/elixir/map_test.exs

```diff
@@ -395,6 +395,14 @@ defmodule MapTest do
     assert quoted == {:%, [], [User, {:%{}, [], [{:foo, 1}]}]}
   end
 
+  test "structs with bitstring defaults" do
+    defmodule WithBitstring do
+      defstruct bitstring: <<255, 127::7>>
+    end
+
+    assert struct!(WithBitstring).bitstring == <<255, 127::7>>
```

Comment:
This check is not needed as even if we use an empty bin in your implementation the value is still the same. Following the issue there should be a check if the module compiles. Optionally there could be an extra test for the new `:elixir_erl.elixir_to_erl/2` function clause.

Reply:
> This check is not needed as even if we use an empty bin in your implementation the value is still the same. 

@Eiji7 good catch thank you! 💜 This was needed by struct_info, fixed.

> Following the issue there should be a check if the module compiles.

This is being checked, otherwise `defmodule` would fail.

> Optionally there could be an extra test for the new :elixir_erl.elixir_to_erl/2 function clause.

I thought the same, but I don't think there is any direct test for `elixir_to_erl` at this moment.